### PR TITLE
Conform to insights api guidelines

### PR DESCRIPTION
**Pull request by mstead**
This PR addresses:
1. Remove camel case property names.
2. Property API definition for Errors wrapping object.
3. Ensure our ObjectMapper will serialize dates to ISO-8601 standard.
4. Pluralize the inventory endpoint

### CODE REVIEW

**Comment by mstead** [comment]
I'm going to add the pluralization of inventory endpoint to this PR. Small change but will have a small conflict in a separate PR.

File: src/test/java/org/candlepin/insights/jackson/ObjectMapperContextResolverTest.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import org.junit.jupiter.api.Test;
+
+import java.util.Calendar;
+import java.util.TimeZone;
+
+
+public class ObjectMapperContextResolverTest {
+
```
Comment (awood) [inline src/test/java/org/candlepin/insights/jackson/ObjectMapperContextResolverTest.java]: Mind dropping a comment here?
```suggestion
/**
 * Ensure that dates are in ISO-8601 format
 */
```

**Comment by awood** [comment]
Looks good other than just a request for a comment denoting the specific format we're expecting.

File: src/test/java/org/candlepin/insights/jackson/ObjectMapperContextResolverTest.java (hunk 1 above)
Comment (mstead) [reply src/test/java/org/candlepin/insights/jackson/ObjectMapperContextResolverTest.java]: Comment added.

**Pull request merged**